jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3TreeTable.kt

```diff
@@ -118,20 +117,22 @@ class S3TreeTable(
         launch {
             try {
                 bucket.download(project, objectNode.key, objectNode.versionId, fileWrapper.file.outputStream())
-                withContext(edt) {
-                    // If the file type is not associated, prompt user to associate. Returns null on cancel
-                    fileWrapper.virtualFile?.let {
-                        ApplicationManager.getApplication().runWriteAction {
-                            it.isWritable = false
-                        }
-                        FileTypeChooser.getKnownFileTypeOrAssociate(it, project) ?: return@withContext
-                        // set virtual file to read only
-                        FileEditorManager.getInstance(project).openFile(it, true, true).ifEmpty {
-                            notifyError(project = project, content = message("s3.open.viewer.failed.unsupported"))
+                if (S3VirtualBucket.bucketExists) {
+                    withContext(edt) {
+                        // If the file type is not associated, prompt user to associate. Returns null on cancel
+                        fileWrapper.virtualFile?.let {
+                            ApplicationManager.getApplication().runWriteAction {
+                                it.isWritable = false
+                            }
+                            FileTypeChooser.getKnownFileTypeOrAssociate(it, project) ?: return@withContext
+                            // set virtual file to read only
+                            FileEditorManager.getInstance(project).openFile(it, true, true).ifEmpty {
+                                notifyError(project = project, content = message("s3.open.viewer.failed.unsupported"))
+                            }
                         }
                     }
+                    S3Telemetry.downloadObject(project, true)
                 }
-                S3Telemetry.downloadObject(project, true)
             } catch (e: Exception) {
```

Comment:
An alternative way to do this is to catch the `nosuchbucketexception` in places we catch a generic `Exception`

Reply:
Looking into this